api/src/Model/Services/AssignData.php

```diff
@@ -85,9 +85,10 @@ function updateDewar($dewarId, $status)
 
     function deactivateDewar($dewarId)
     {
-        $this->updateDewarHistory($dewarId, 'unprocessing');
+        $location = $this->db->pq("SELECT storagelocation FROM dewar WHERE dewarid=:1", array($dewarId));
```

Comment:
I think that dewar.storagelocation can be null, maybe do a check here and set it to blank if it is null?

Reply:
Also line 105 look like it could be null if the beamline is not set on the container

Reply:
I've added a check just before the insert statement (line 105)